src/render.py

```diff
@@ -61,16 +61,29 @@ def render_pep440_branch(pieces):
     return rendered
 
 
+def pep440_split_post(ver):
+    """Split pep440 version string at the post-release segment.
+
+    Returns the release segments before the post-release and the
+    post-release version number (or -1 if no post-release segment is present).
+    """
+    vc = str.split(ver, ".post")
+    return vc[0], int(vc[1] or 0) if len(vc) == 2 else -1
+
+
 def render_pep440_pre(pieces):
-    """TAG[.post0.devDISTANCE] -- No -dirty.
+    """TAG[.postN.devDISTANCE] -- No -dirty.
 
     Exceptions:
     1: no tags. 0.post0.devDISTANCE
     """
     if pieces["closest-tag"]:
-        rendered = pieces["closest-tag"]
-        if pieces["distance"]:
-            rendered += ".post0.dev%d" % pieces["distance"]
+        tag_version, post_version = pep440_split_post(pieces["closest-tag"])
+        rendered = tag_version
+        if post_version > -1 or pieces["distance"]:
```

Comment:
This should be an `and`, if there have been no commits the closest-tag should be used as is, otherwise the pre-release dev segment and incremented post release segment need adding.